src/components/AdjMatrix/AdjMatrixMethods.ts

```diff
@@ -505,39 +505,24 @@ export class View {
    * @return none
    */
   private appendEdgeLabels(): void {
-    // constants for manipulating the x and y values of the different clip parts
-    const clipXValue = 74;
-    const clipYValue = 10;
-
-    // clip path for row labels
-    select('#matrix').append('clipPath').attr('id', 'text-clip');
-
-    // the shape of the clip path
-    select('#text-clip')
-      .append('rect')
-      .attr('x', -(clipXValue + 6))
-      .attr('y', -clipYValue)
-      .attr('width', clipXValue)
-      .attr('height', 600);
-
-    this.edgeRows
-      .append('text')
-      .attr('clip-path', 'url(#text-clip)')
-      .attr('class', 'rowLabel')
-      .attr('id', (d: Node) => `rowLabel${d.id}`)
-      .attr('z-index', 30)
-      .attr('x', -clipXValue)
-      .attr('y', clipYValue / 2)
-      .attr('dy', '.75em')
-      .attr('text-anchor', 'start')
-      .style('font-size', `${this.nodeFontSize}pt`)
-      .text((d: Node) => d._key)
-      .on('mouseover', (d: Node) => this.hoverNode(d.id))
-      .on('mouseout', (d: Node) => this.unHoverNode(d.id))
-      .on('mouseover', (d: Node, i: number, nodes: any) => {
-        this.showToolTip(d, i, nodes);
-        this.hoverNode(d.id);
-      })
+    const labelContainerHeight = 25;
+    const rowLabelContainerStart = 75;
+    const labelContainerWidth = labelContainerHeight + 60;
+
+    // add foreign objects for label
+    const edgeRowForeignObject = this.edgeRows
+      .append('foreignObject')
+      .attr('x', -rowLabelContainerStart)
+      .attr('y', -5)
+      .attr('width', labelContainerWidth)
+      .attr('height', labelContainerHeight);
+
+    const rowLabelBody = edgeRowForeignObject.append('xhtml:body');
+
+    const rowLabel = rowLabelBody
+      .append('p')
```

Comment:
You don't need to add a body each time, you can just add the p
```suggestion
      .attr('height', labelContainerHeight);

    const rowLabel = edgeRowForeignObject
      .append('xhtml:p')
```